src/System.IO.Pipelines/PipeFactory.cs

```diff
@@ -45,13 +45,22 @@ public IPipeReader CreateReader(Stream stream)
             }
 
             var pipe = new Pipe(_pool);
-            ExecuteCopyToAsync(pipe, stream);
+            var ignore = ExecuteCopyToAsync(pipe, stream);
             return pipe;
         }
 
-        private Task ExecuteCopyToAsync(Pipe pipe, Stream stream)
+        private async Task ExecuteCopyToAsync(Pipe pipe, Stream stream)
         {
-            return stream.CopyToAsync(pipe);
+            try
+            {
+                await stream.CopyToAsync(pipe);
+            }
+            catch (Exception ex)
+            {
+                pipe.Writer.Complete(ex);
+                return;
+            }
+            pipe.Writer.Complete();
```

Comment:
@Drawaes. I just saw that you did change StreamExtensions.CopyToAsync() to call Complete() and swallow any exceptions. In that case both the try/catch and the calls to Complete() in this method seem redundant.